### PR TITLE
logging: Improve flowbit logs

### PR DESCRIPTION
**Pull request by Riju19**
Adds debugging logs for flowbit requirements and fixes Optimization #3205

Make sure these boxes are signed before submitting your Pull Request
-- thank you.

- [x] I have read the contributing guide lines at
  https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation
  contribution agreement at
  https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the
  changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3205

Describe changes:
Improves flowbit handling logs in the debug mode

### CODE REVIEW

**Comment by inashivb** [comment]
Hi, Riju!

Thanks for your first contribution to the project! :)
Looks good to me. @jasonish do you like this way of logging? The person who reported this as a problem liked this.

Good job!

**Comment by jasonish** [comment]
> Looks good to me. @jasonish do you like this way of logging? The person who reported this as a problem liked this.

Yeah, looks good.  One comment with respect to naming inline.

File: suricata/update/main.py [hunk 1]
```diff
@@ -619,7 +619,9 @@ def dump_sample_configs():
 def resolve_flowbits(rulemap, disabled_rules):
     flowbit_resolver = rule_mod.FlowbitResolver()
     flowbit_enabled = set()
+    round = 1
     while True:
+        logger.debug("Checking flowbits for round %s of rules.", round)
```
Comment (jasonish) [inline suricata/update/main.py:624]: What we're doing here is generally called "multi-pass". So I think we should use the work pass instead.
```
Checking flowbits for pass %d of rules."
```

File: suricata/update/main.py [hunk 2]
```diff
@@ -619,7 +619,9 @@ def dump_sample_configs():
...
         flowbits = flowbit_resolver.get_required_flowbits(rulemap)
         logger.debug("Found %d required flowbits.", len(flowbits))
```
Comment (jasonish) [inline suricata/update/main.py:626]: Alternatively we could reword this:
```
Found %d required flowbits (pass %d)
```

File: suricata/update/main.py (hunk 1 above)
Comment (Riju19) [reply suricata/update/main.py]: I have incorporated the suggested change.
But I would like to know a little more about what you mean by multi-pass and work-pass.

Comment (jasonish) [reply suricata/update/main.py]: Oops, that was a type.  I meant to write "word", not "work".

File: suricata/update/main.py [hunk 3]
```diff
@@ -619,9 +619,11 @@ def dump_sample_configs():
 def resolve_flowbits(rulemap, disabled_rules):
     flowbit_resolver = rule_mod.FlowbitResolver()
     flowbit_enabled = set()
+    pass_ = 1
     while True:
+        logger.debug("Checking flowbits for pass %d of rules.", pass_)
         flowbits = flowbit_resolver.get_required_flowbits(rulemap)
-        logger.debug("Found %d required flowbits.", len(flowbits))
+        logger.debug("Found %d required flowbits (pass %d).", len(flowbits), pass_)
```
Comment (jasonish) [inline suricata/update/main.py:626]: Remove this line please... And, see next comment...

File: suricata/update/main.py [hunk 4]
```diff
@@ -619,9 +619,11 @@ def dump_sample_configs():
...
         required_rules = flowbit_resolver.get_required_rules(rulemap, flowbits)
         logger.debug(
             "Found %d rules to enable to for flowbit requirements",
```
Comment (jasonish) [inline suricata/update/main.py:629]: Update this line to include the pass number... For example:
```
Found %d rules to enable to for flowbit requirements (pass %d)
```

Gives us a little cleaner output.

**Review by jasonish — CHANGES_REQUESTED** [review]
Please follow out guidelines of creating a new PR for the next pull request.

Thanks!

File: suricata/update/main.py (hunk 4 above)
Comment (Riju19) [reply suricata/update/main.py:629]: Just a tiny doubt about the semantics of the log here - would this be better worded as :
```
Found %d rules to enable for flowbit requirements (pass %d)
```
(I just removed the second 'to')

Comment (jasonish) [reply suricata/update/main.py:629]: Yes, that is better.  Thanks.

**Comment by jasonish** [comment]
Current PR: https://github.com/OISF/suricata-update/pull/265

**Pull request closed without merging**